packages/network/src/logic/Node.ts

```diff
@@ -308,8 +311,25 @@ export class Node extends EventEmitter {
 
         if (isUnseen) {
             logger.trace('received from %s data %j', source, streamMessage.messageId)
+
+            let groupKeyPropagationTargets
+            if (this.acceptProxyConnections) {
+                if (GroupKeyRequest.is(streamMessage)) {
+                    const { recipient } = GroupKeyRequest.fromStreamMessage(streamMessage) as GroupKeyRequest
+                    groupKeyPropagationTargets = this.proxyStreamConnectionManager.getAllConnectionNodeIds(streamPartId, recipient)
+                } else if (GroupKeyResponse.is(streamMessage)) {
+                    const { recipient } = GroupKeyResponse.fromStreamMessage(streamMessage) as GroupKeyResponse
+                    groupKeyPropagationTargets = this.proxyStreamConnectionManager.getAllConnectionNodeIds(streamPartId, recipient)
+                }
+            } else if (
+                this.streamPartManager.isBehindProxy(streamMessage.getStreamPartID())
+                && this.proxyStreamConnectionManager.isProxiedStreamPart(streamMessage.getStreamPartID(), ProxyDirection.SUBSCRIBE)
+            ) {
+                groupKeyPropagationTargets = [...this.streamPartManager.getInboundNodesForStreamPart(streamPartId)]
+            }
```

Comment:
Extract method and include the normal neighbors also as part of the `Propagation` refactor.